packages/react/src/components/NumberInput/NumberInput.tsx

```diff
@@ -372,10 +372,14 @@ const NumberInput = React.forwardRef<HTMLInputElement, NumberInputProps>(
           getDecimalPlaces(step)
         );
         const floatValue = parseFloat(rawValue.toFixed(precision));
-        const newValue =
-          typeof min !== 'undefined' && typeof max !== 'undefined'
-            ? Math.min(Math.max(floatValue, min), max)
-            : floatValue;
+        let newValue = floatValue;
+        if (typeof min !== 'undefined') {
+          newValue = Math.max(newValue, min);
+        }
+        if (typeof max !== 'undefined') {
+          newValue = Math.min(newValue, max);
+        }
```

Comment:
This functionality is similar to `clamp` in packages/react/src/components/Slider/Slider.tsx. Would it be better to abstract a clamp function into a utils file (i.e., packages/react/src/internal/utils.ts) and import it for use here and in packages/react/src/components/Slider/Slider.tsx (deleting the `clamp` function that's already in that file)?

```tsx
const clamp = (value: number, min?: number, max?: number) =>
  Math.min(max ?? Infinity, Math.max(min ?? -Infinity, value));
```

```suggestion
        const newValue = clamp(floatValue, min, max);
```

Reply:
thanks updated!